spi_nand_flash/src/dhara_glue.c

```diff
@@ -235,10 +246,17 @@ int dhara_nand_read(const struct dhara_nand *n, dhara_page_t p, size_t offset, s
 
     ESP_GOTO_ON_ERROR(read_page_and_wait(dev, p, &status), fail, TAG, "");
 
-    if (is_ecc_error(status)) {
-        ESP_LOGD(TAG, "read ecc error, page=%"PRIu32"", p);
+    ecc_status_t bits_corrected_status = get_ecc_status(status, dev->ecc_data.ecc_status_reg_len_in_bits);
+    dev->ecc_data.ecc_corrected_bits_status = bits_corrected_status;
+    if (bits_corrected_status) {
         dhara_set_error(err, DHARA_E_ECC);
-        return -1;
+        if (bits_corrected_status == STAT_ECC_MAX) {
+            ESP_LOGE(TAG, "%s: Error while initializing value of ecc_status_reg_len_in_bits", __func__);
+            return -1;
+        } else if (bits_corrected_status == STAT_ECC_NOT_CORRECTED) {
+            ESP_LOGD(TAG, "read ecc error, page=%"PRIu32"", p);
+            return -1;
```

Comment:
According to

https://github.com/dlbeer/dhara/blob/1b166e41b74b4a62ee6001ba5fab7a8805e80ea2/dhara/nand.h#L89-L90

we should set DHARA_E_ECC error if an uncorrectable error occurs. If i'm reading the code right, now we also set it also if e.g. one bit error occurred and was corrected.

Reply:
My understanding of the statement is that "if an uncorrectable error occurs, it should return -1 and set err to E_ECC." However, if there’s an ECC error with corrected bits, we are setting E_ECC in err but not returning -1 and this error is used to decide if a data refresh is needed after successful nand_read.  https://github.com/espressif/idf-extra-components/pull/437/files#diff-41c825f56b23bffd93941478281fd19aba3f9ddedda4b61af5c319827d0bdf1eL306

Reply:
FYI, a small update was made to the is_ecc_error() API. It’s not related to the above comment—just a restructuring.

Reply:
Updated. Now DHARA_E_ECC error will be set only if uncorrectable error occurs.